packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -29,28 +33,48 @@ class ImgElementPlatformView extends StatelessWidget {
   static const String _viewType = 'Flutter__ImgElementImage__';
   static bool _registered = false;
 
+  /// Override this to provide a custom implementation of [ui_web.platformViewRegistry.registerViewFactory].
+  ///
+  /// This should only be used for testing.
+  @visibleForTesting
+  static RegisterViewFactory? debugOverrideRegisterViewFactory;
+  static RegisterViewFactory get _registerViewFactory =>
+      debugOverrideRegisterViewFactory ?? ui_web.platformViewRegistry.registerViewFactory;
+
   static void _register() {
     assert(!_registered);
     _registered = true;
-    ui_web.platformViewRegistry.registerViewFactory(_viewType, (int viewId, {Object? params}) {
+    _registerViewFactory(_viewType, (int viewId, {Object? params}) {
       final Map<Object?, Object?> paramsMap = params! as Map<Object?, Object?>;
       // Create a new <img> element. The browser is able to display the image
       // without fetching it over the network again.
       final web.HTMLImageElement img = web.document.createElement('img') as web.HTMLImageElement;
       img.src = paramsMap['src']! as String;
+      img.style.width = '100%';
+      img.style.height = '100%';
+      final String? fit = paramsMap['fit'] as String?;
+      if (fit != null) {
+        img.style.objectFit = fit;
+      }
       return img;
     });
   }
 
   /// The `src` URL for the `<img>` tag.
   final String? src;
 
+  /// The `object-fit` CSS property for the `<img>` tag.
+  final web.CSSObjectFit? fit;
```

Comment:
Can you explain why the new type `CSSObjectFit` is needed? It seems to me that in an actual app, this type is only used for intermediate type passthrough: a `BoxFit` is converted to a `CSSObjectFit`, which is then converted to a string. Can we use `BoxFit` all the way until it's converted to a CSS string?